org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/LaunchingPlugin.java

```diff
@@ -582,6 +589,66 @@ public void saving(ISaveContext context1) throws CoreException {
 		ResourcesPlugin.getWorkspace().addResourceChangeListener(this, IResourceChangeEvent.PRE_DELETE | IResourceChangeEvent.PRE_CLOSE);
 		DebugPlugin.getDefault().getLaunchManager().addLaunchListener(this);
 		DebugPlugin.getDefault().addDebugEventListener(this);
+		IEclipsePreferences instanceNode = InstanceScope.INSTANCE.getNode(getBundle().getSymbolicName());
+		IEclipsePreferences defaultNode = DefaultScope.INSTANCE.getNode(getBundle().getSymbolicName());
+		if (instanceNode.getBoolean(PREF_DETECT_VMS_AT_STARTUP, defaultNode.getBoolean(PREF_DETECT_VMS_AT_STARTUP, true))) {
+			new Job(LaunchingMessages.lookupInstalledJVMs) {
```

Comment:
Can code from this job be extracted in a separate class?

Reply:
Yes, I'll do that tomorrow.